js/src/Themes.js

```diff
@@ -50,10 +55,35 @@ export class ThemeModel extends WidgetModel {
     } else if (document.body.classList.contains("theme-light")) {
       this.set("dark", false);
       this.save_changes();
+    } else if (window.Jupyter) {
+      // Special case for Jupyter Notebook
+      this.set("dark", false);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-dark") {
+      // Special case for VS Code
+      vuetify.framework.theme.dark = true;
+      this.set("dark", true);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-light") {
+      vuetify.framework.theme.dark = false;
+      this.set("dark", false);
+      this.save_changes();
+    } else if (document.documentElement.matches("[theme=dark]")) {
+      vuetify.framework.theme.dark = true;
+      this.set("dark_effective", true);
+      this.save_changes();
+    } else if (document.documentElement.matches("[theme=light]")) {
+      vuetify.framework.theme.dark = false;
+      this.set("dark_effective", false);
+      this.save_changes();
+    } else {
+      let osPrefersDark =
+        window.matchMedia &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches;
+      vuetify.framework.theme.dark = osPrefersDark;
+      this.set("dark_effective", osPrefersDark);
+      this.save_changes();
```

Comment:
I think in addition to this, we should then also watch:
```
                window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
                    this.setTheme();
                });
```

We could always listen to this event btw, we don't have to do that conditionally (it will trigger, but not have an effect on non-solara systems)